### ISSUE TITLE
new ObjectID error with a String

### ISSUE THREAD

**Issue by AlejandroArmenteros** [Automattic/mongoose#4877]
Hi!
I have an error when i try to findById a document. I have this in my collection users:
```
{
 "_id" : ObjectId("586f7bbe2164611aaac549e9"), 
 "username" : "prueba",
  "__v" : 0 
}
```
I get the id in String, so i have "586f7bbe2164611aaac549e9".
I read that i must to get the ObjectId with: 

`User.findById(ObjectId("586f7bbe2164611aaac549e9"), .......`

And i get this error

```
TypeError: hex is not a function
    at Function.from (native)
    at Function.from (native)
    at new ObjectID (......./node_modules/bson/lib/bson/objectid.js:52:32)
    at ObjectID (..../node_modules/bson/lib/bson/objectid.js:31:42)
    at exports.ensureAuthenticated (..../src/middleware/auth.js:29:17)
    at Layer.handle [as handle_request] (..../node_modules/express/lib/router/layer.js:95:5)
    at next (..../node_modules/express/lib/router/route.js:131:13)
    at Route.dispatch (..../node_modules/express/lib/router/route.js:112:3)
    at Layer.handle [as handle_request] (..../node_modules/express/lib/router/layer.js:95:5)
    at /..../node_modules/express/lib/router/index.js:277:22
    at Function.process_params (..../node_modules/express/lib/router/index.js:330:12)
    at next (..../node_modules/express/lib/router/index.js:271:10)
    at Function.handle (..../node_modules/express/lib/router/index.js:176:3)
    at router (..../node_modules/express/lib/router/index.js:46:12)
    at Layer.handle [as handle_request] (..../node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (..../node_modules/express/lib/router/index.js:312:13)
```

I check before ObjectId.isValid(id), and it return me true.
I don't know if is a bug or i'm doing something really bad :S

**Comment by damen-dd** [comment]
I have the exact same issue.

**Comment by ahmadferdous** [comment]
I also faced the same error while using mongoose. This is caused by issue mongodb/js-bson#204 and a fix mongodb/js-bson#205 has been submitted

**Comment by varunjayaraman** [comment]
This will hopefully be resolved when that PR is merged in in js-bson. The solution is to either use node >= v6.0 or to use mongoose <= 4.7.2. I'm gonna close this since this is also a duplicate of https://github.com/Automattic/mongoose/issues/4867
